datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -748,6 +779,25 @@ mod tests {
         projection: Option<Vec<usize>>,
         statistics: Statistics,
         table_partition_cols: Vec<(String, DataType)>,
+    ) -> FileScanConfig {
+        let table_partition_cols = table_partition_cols
+            .iter()
+            .map(|(name, dtype)| Field::new(name, dtype.clone(), false))
+            .collect::<Vec<_>>();
+
+        config_for_proj_with_field_tab_part(
+            file_schema,
+            projection,
+            statistics,
+            table_partition_cols,
+        )
+    }
+
+    fn config_for_proj_with_field_tab_part(
```

Comment:
I find this name confusing given the three letter abbreviations and I don't think this is common elsewhere in the DataFusion codebase. 

How about something like

```suggestion
    fn config_for_projection_with_partition_fields(
```

Or maybe instead you could change `config_for_projection` to take `table_partition_cols: Vec<Field>,` and make a function like

```rust
/// Convert all 
fn partition_cols(  table_partition_cols: Vec<(&str, DataType)>) -> Vec<Field> {
         table_partition_cols
            .iter()
            .map(|(name, dtype)| Field::new(name, dtype.clone(), false))
            .collect::<Vec<_>>()
}
```


And then convert the call sites of `config_for_projection` to be `config_for_projection(.., partition_cols(..))`

Reply:
I implemented your second suggestion @alamb . Thanks